### PR TITLE
Update the Array.prototype.reverse method to support ES6 version

### PR DESCRIPTION
**Pull request by szilagyiadam**
JerryScript-DCO-1.0-Signed-off-by: Adam Szilagyi aszilagy@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c [hunk 1]
```diff
@@ -610,72 +589,112 @@ ecma_builtin_array_prototype_object_reverse (ecma_value_t this_arg, /**< this ar
...
+      goto str_clean_up;
     }
 #endif /* ENABLED (JERRY_ES2015_BUILTIN_PROXY) */
 
     bool lower_exist = ecma_is_value_true (has_lower);
     bool upper_exist = ecma_is_value_true (has_upper);
 
-    /* 6.h */
+    ecma_value_t lower_value = ECMA_VALUE_EMPTY;
+    ecma_value_t upper_value = ECMA_VALUE_EMPTY;
```
Comment (galpeter) [inline jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c:600]: It seems that ES2015 and ES5.1 differs in the order of things to do. Probably we should create two methods to processes the upper/lower values in es6 and in es51.

File: jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c [hunk 2]
```diff
@@ -610,72 +589,112 @@ ecma_builtin_array_prototype_object_reverse (ecma_value_t this_arg, /**< this ar
...
+
+#if ENABLED (JERRY_ES2015)
+    if (lower_exist)
```
Comment (galpeter) [inline jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c:603]: In such cases I think we should only guard the `if` part and leave the opening a closing brackets alone.

File: jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c [hunk 3]
```diff
@@ -605,6 +589,18 @@ ecma_builtin_array_prototype_object_reverse (ecma_value_t this_arg, /**< this ar
...
 
+    bool lower_exist = ecma_is_value_true (has_lower);
+
+    if (lower_exist)
+    {
+      lower_value = ecma_op_object_get (obj_p, lower_str_p);
+
+      if (ECMA_IS_VALUE_ERROR (lower_value))
+      {
+        goto clean_up;
```
Comment (galpeter) [inline jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c:600]: In these error cases, should we return the error for the user? Do we have test cases for these?

Comment (szilagyiadam) [reply jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c:600]: These test cases have already been tested in the ES5 tests, this update just changes the order of methods, but the behaviour when one of these error occurs is the same.

**Pull request merged**